### PR TITLE
Add vision detect_labels() system tests.

### PR DESCRIPTION
**Pull request by daspecster**
Towards #2753 

Adds system tests for `detect_labels()`.

```bash
test_detect_faces_content (vision.TestVisionClientFace) ... ok
test_detect_faces_filename (vision.TestVisionClientFace) ... ok
test_detect_faces_gcs (vision.TestVisionClientFace) ... ok
test_detect_labels_content (vision.TestVisionClientLabel) ... ok
test_detect_labels_filename (vision.TestVisionClientLabel) ... ok
test_detect_labels_gcs (vision.TestVisionClientLabel) ... ok
test_detect_logos_content (vision.TestVisionClientLogo) ... ok
test_detect_logos_filename (vision.TestVisionClientLogo) ... ok
test_detect_logos_gcs (vision.TestVisionClientLogo) ... ok

----------------------------------------------------------------------
Ran 9 tests in 12.904s

OK
________________________________________________________ summary ________________________________________________________
  system-tests: commands succeeded
  congratulations :)
```

### CODE REVIEW

File: system_tests/vision.py [hunk 1]
```diff
@@ -54,7 +55,15 @@ def tearDownModule():
     bucket_retry(Config.TEST_BUCKET.delete)(force=True)
 
 
-class TestVisionClientLogo(unittest.TestCase):
+class BaseVisionTestCase(unittest.TestCase):
+    def _assert_coordinate(self, coordinate):
+        if coordinate is None:
+            return
+        self.assertIsInstance(coordinate, (int, float))
+        self.assertGreater(abs(coordinate), 0.0)
```
Comment (dhermes) [inline system_tests/vision.py]: Why didn't this become `self.assertNotEqual(coordinate, 0.0)`?

File: system_tests/vision.py [hunk 2]
```diff
@@ -54,7 +55,15 @@ def tearDownModule():
...
+
+
+class TestVisionClientLogo(BaseVisionTestCase):
```
Comment (dhermes) [inline system_tests/vision.py]: If this class doesn't use the methods in the base, it doesn't need to inherit from it

File: system_tests/vision.py [hunk 3]
```diff
@@ -215,3 +218,55 @@ def test_detect_faces_filename(self):
...
+class TestVisionClientLabel(BaseVisionTestCase):
+    def setUp(self):
+        self.to_delete_by_case = []
+
+    def tearDown(self):
+        for value in self.to_delete_by_case:
+            value.delete()
+
+    def _assert_label(self, label):
+        descriptions = ['car', 'vehicle', 'land vehicle', 'automotive design',
```
Comment (dhermes) [inline system_tests/vision.py]: Make `descriptions` a class constant

File: system_tests/vision.py [hunk 4]
```diff
@@ -215,3 +218,55 @@ def test_detect_faces_filename(self):
...
+                        'wheel', 'automobile make', 'luxury vehicle',
+                        'sports car', 'performance car', 'automotive exterior']
+        self.assertIsInstance(label, EntityAnnotation)
+        self.assertIn(label.description, descriptions)
+        self.assertIsInstance(label.mid, unicode)
```
Comment (dhermes) [inline system_tests/vision.py]: Use `six.text_types`

**Comment by daspecster** [comment]
Change made and squashed.

**Pull request merged**